### PR TITLE
Add support for requiring err before sending

### LINKED ISSUE #505: Assert that linked ramps check all ports are connected before consuming data
When starting a source we wait until `out` is connected before consuming data.

For linked ramps, we should extend this to `in`. The reason is that otherwise, we might run into a race condition where we get a first event before our `in` port is connected and lose the reply.

We also should allow the user to specify to wait for `err` this way it can be ensured that all errors are captured. This could not be mandatory since a user might be OK with dropping errors?

### PR DESCRIPTION
**Pull request by Licenser**
Signed-off-by: Heinz N. Gies <heinz@licenser.net>

## Description

Introduces the `err_required` field to sources to allow users to require both `out` and `err` to be connected before
the source is started.

## Related

* Related Issues: closed #505 
* Related [docs PR](https://github.com/tremor-rs/tremor-www-docs/pull/55)

## Checklist

* [x] The RFC, if required, has been submitted and approved
* [x] Any user-facing impact of the changes is reflected in docs.tremor.rs
* [x] The code is tested
* [x] Use of unsafe code is reasoned about in a comment

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -50,6 +50,8 @@ pub struct OnRamp {
     #[serde(rename = "linked", default = "Default::default")]
     // TODO validate that this is turned on only for supported onramps (rest, ws)
     pub(crate) is_linked: bool,
+    #[serde(rename = "error_required", default = "Default::default")]
```
Comment (mfelsche) [inline src/config.rs:53]: can we stick to `err` instead of `error`? It might be confusing for people which one to use.

File: src/onramp.rs [hunk 2]
```diff
@@ -43,17 +43,18 @@ pub enum Msg {
 
 pub type Addr = async_channel::Sender<Msg>;
 
+pub(crate) struct OnrampConfig<'cfg> {
```
Comment (mfelsche) [inline src/onramp.rs:46]: 🤝

File: src/repository/artefact.rs [hunk 3]
```diff
@@ -448,7 +449,7 @@ impl Artefact for OnrampArtefact {
                 expect_answers -= 1;
             }
 
-            info!("Onramp {} unklinked.", id);
```
Comment (mfelsche) [inline src/repository/artefact.rs:451]: `unklinked` also has a ring to it, i have to admit :)

File: src/config.rs (hunk 1 above)
Comment (Licenser) [reply src/config.rs:53]: good catch!

**Review by mfelsche — APPROVED** [review]

**Pull request merged**